oak-search-elastic/src/main/java/org/apache/jackrabbit/oak/plugins/index/elastic/ElasticPropertyDefinition.java

```diff
@@ -50,6 +61,14 @@ public ElasticPropertyDefinition(IndexDefinition.IndexingRule idxDefn, String no
                     getOptionalValue(defn, PROP_CANDIDATES, DEFAULT_CANDIDATES));
         }
         this.useInFullTextQuery = this.dynamicBoost && getOptionalValue(defn, PROP_USE_IN_FULL_TEXT_QUERY, true);
+        boolean flattened = getOptionalValue(defn, PROP_IS_FLATTENED, PROP_IS_FLATTENED_DEFAULT);
+        if (analyzed) {
+            // if analyzed is enabled, then flattened needs to be disabled,
+            // because flattened types do not support fulltext queries
```

Comment:
should we also add `nodeScopeIndex` to this exception ? As per the oak docs 

```
nodeScopeIndex
Control whether the value of a property should be part of fulltext index. That is, you can do a jcr:contains(., 'foo') and it will return nodes that have a string property that contains the word foo. Example
/jcr:root/content//element(*, app:Asset)[jcr:contains(., 'image')]

```

Reply:
First I thought so, yes. But actually it's not needed: nodeScopeIndex adds the text to the ":fulltext" field, and that one is not flattened (doesn't need to be).